### PR TITLE
Internal improvement: From Web3Shim to InterfaceAdapter – Step Two: getTransaction

### PR DESCRIPTION
**Pull request by CruzMolina**
Continuation of #2473 that implements Step 2 for `getTransaction` as outlined [here](https://gist.github.com/gnidan/a55bbec28800f64d487054be18730691#file-step2-ts).

### CODE REVIEW

File: packages/interface-adapter/lib/interface-adapter.ts [hunk 1]
```diff
@@ -46,7 +49,11 @@ export class InterfaceAdapter {
     return this.adapter.getBlock(block);
   }
 
-  public setProvider(provider: Provider) {
+  public setProvider(provider: Provider): void {
```
Comment (gnidan) [inline packages/interface-adapter/lib/interface-adapter.ts]: Can we make this async? I know that was last PR (sorry!)

File: packages/interface-adapter/lib/interface-adapter.ts [hunk 2]
```diff
@@ -46,7 +48,11 @@ export class InterfaceAdapter {
     return this.adapter.getBlock(block);
   }
 
-  public setProvider(provider: Provider) {
+  public setProvider(provider: Provider): Promise<void> {
```
Comment (gnidan) [inline packages/interface-adapter/lib/interface-adapter.ts]: is `adapter.setProvider` not being used? we'll need to add `await` to those (this is the part that I figured you might object to :)

Comment (CruzMolina) [reply packages/interface-adapter/lib/interface-adapter.ts]: we only use it in `@truffle/contract`'s `setProvider` method. I believe at least one of the CI jobs would be failing if it wasn't properly resolving.

Comment (CruzMolina) [reply packages/interface-adapter/lib/interface-adapter.ts]: anddd it's gone.

**Pull request merged**